File: src/main/java/no/novari/flyt/authorization/client/sourceapplications/model/SiDESourceApplication.java
package no.novari.flyt.authorization.client.sourceapplications.model;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class SiDESourceApplication extends BaseSourceApplication {
    public SiDESourceApplication(
            @Value("${fint.flyt.side.sso.client-id:#{null}}") String clientId,
            @Value("${fint.flyt.side.available:false}") boolean available
    ) {
        super(9L, "SiDE", clientId, available);
    }
}

File: src/main/kotlin/no/novari/flyt/authorization/client/sourceapplications/model/SiDESourceApplication.kt
package no.novari.flyt.authorization.client.sourceapplications.model

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class SiDESourceApplication(
    @Value("\${fint.flyt.side.sso.client-id:#{null}}")
    clientId: String?,
    @Value("\${fint.flyt.side.available:false}")
    available: Boolean,
) : BaseSourceApplication(
        id = 9L,
        displayName = "SiDE",
        clientId = clientId,
        available = available,
    )
